modules/system-disaster-recovery/src/main/java/org/apache/ignite/internal/disaster/system/SystemDisasterRecoveryStorage.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.disaster.system;
+
+import org.apache.ignite.internal.cluster.management.ClusterState;
+import org.apache.ignite.internal.disaster.system.message.ResetClusterMessage;
+import org.apache.ignite.internal.disaster.system.storage.ClusterResetStorage;
+import org.apache.ignite.internal.lang.ByteArray;
+import org.apache.ignite.internal.util.ByteUtils;
+import org.apache.ignite.internal.vault.VaultEntry;
+import org.apache.ignite.internal.vault.VaultManager;
+import org.jetbrains.annotations.Nullable;
+
+class SystemDisasterRecoveryStorage extends ClusterResetStorage {
```

Comment:
I find it a little bit of an overkill that `ClusterResetStorage` is a class and not an interface. Can we make it an interface and have the whole implementation here?

Reply:
If `ClusterResetStorage` is an interface and `SystemDisasterRecoveryStorage` is its implementation, we can't instantiate the latter in `CMGManager` because `SystemDisasterRecoveryStorage` is in a module that depends on the CMG module.

* If we move `SystemDisasterRecoveryStorage` to the system-disaster-recovery-api module, this module will need to have access to CMG messages. To make them accessible, we would have to introduce another module, cmg-api.
* Or we could pass just the interface to the CMGManager, so I would have to modify all tests instantiating CMGManager, and probably also add a simple test implementation of the interface.

On the other hand, with current approach (which is just a bit smelly), CMGManager instantiates the storage itself, and no new modules need to be added.

Do you have other suggestions?

Reply:
> Or we could pass just the interface to the CMGManager, so I would have to modify all tests instantiating CMGManager, and probably also add a simple test implementation of the interface.

Yes, let's have `ClusterResetStorage` as an interface, pass it to CmgManager's constructor and have  `SystemDisasterRecoveryStorage` as its production implementation. You will need to update all tests, that's true, but you won't have to introduce a separate implementation, because you will still be able to use `SystemDisasterRecoveryStorage` in tests.